gossip-base/src/main/java/org/apache/gossip/event/data/DataEventManager.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gossip.event.data;
+
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+
+public class DataEventManager {
+    private final List<UpdateNodeDataEventHandler> perNodeDataHandlers;
+    private final BlockingQueue<Runnable> perNodeDataHandlerQueue;
+    private final ExecutorService perNodeDataEventExecutor;
+
+    private final List<UpdateSharedDataEventHandler> sharedDataHandlers;
+    private final BlockingQueue<Runnable> sharedDataHandlerQueue;
+    private final ExecutorService sharedDataEventExecutor;
+
+
+    public DataEventManager() {
```

Comment:
This class should be constructed with the MetricRegistry so that counters are recorded.